### PR TITLE
refactor: changed to typeorem

### PR DESCRIPTION
**Pull request by Nikajighauri**

### CODE REVIEW

File: src/users/entities/users.entity.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+@Entity()
+export class User {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  email: string;
+
+  @Column()
+  password: string;
```
Comment (pailodze) [inline src/users/entities/users.entity.ts:12]: We need createdAt deletedAt udpatedAt

File: src/users/repositories/users.repository.ts [hunk 2]
```diff
@@ -1,52 +1,36 @@
...
+  async update(id: number, updateUsersDto: UpdateUsersDto): Promise<User> {
+    await this.usersRepository.update(id, updateUsersDto);
+    return await this.findOne(id);
   }
 
-  remove(id: number): UserInterface[] {
-    const user: FindOneUserInterface = this.findOne(id);
-    return this.users.splice(user.index, 1);
+  async remove(id: number): Promise<DeleteResult> {
+    return await this.usersRepository.delete(id);
```
Comment (pailodze) [inline src/users/repositories/users.repository.ts:34]: softRemove

File: src/users/users.controller.ts [hunk 3]
```diff
@@ -7,40 +7,41 @@ import {
...
-    return this.usersService.findAll();
+  async findAll(): Promise<User[]> {
+    return await this.usersService.findAll();
   }
 
   @Get(':id')
-  findOne(@Param('id') id: string): UserInterface {
-    return this.usersService.findOne(+id);
+  async findOne(@Param('id') id: string): Promise<User> {
+    return await this.usersService.findOne(+id);
```
Comment (pailodze) [inline src/users/users.controller.ts:32]: Use Number() for casting nubmer

**Review by pailodze — CHANGES_REQUESTED** [review]

File: src/users/entities/users.entity.ts (hunk 1 above)
Comment (beqagerliAni) [reply src/users/entities/users.entity.ts:12]: ok

**Pull request merged**